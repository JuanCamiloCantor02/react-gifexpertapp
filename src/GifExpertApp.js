import React from 'react';
import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifCollection } from './components/GifCollection';


export const GifExpertApp = ({defaultCategories = []}) =>{
    /* const categories = ["Dragon ball Z", "Pokemon", "Un Show Mas"] */

    /* tener una categoria en duro
    const [categories, setCategories] = useState(["venom"]); */
    const [categories, setCategories] = useState(defaultCategories);
    
    /* const handleAdd = () =>{
          setCategories([...categories, "Mr Beats"]);
    } */
    
    return(
      <>
        <h1>GifExpertApp</h1>
        <AddCategory setCategories={setCategories} />
        <hr />

        

       <ol>
           {
               categories.map(category =>{
                   return <GifCollection key={category} category={category} />

               })
           }
       </ol>
  
      </>
    );
  }
  


