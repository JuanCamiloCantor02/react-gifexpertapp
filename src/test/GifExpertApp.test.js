import React from 'react'
import {shallow} from "enzyme";
import { GifExpertApp } from '../GifExpertApp';


describe('pruebas en <GifExpertApp/>', () => {
    
    test('debe mostrarse correctamente ', () => {
        
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe mostrar una lista de categorias', () => {
        
        const categories = ["Batman", "Goku"]
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifCollection").length).toBe(categories.length);

    })
    

})
