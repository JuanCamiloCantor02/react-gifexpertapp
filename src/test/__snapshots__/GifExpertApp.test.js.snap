// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp/> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifCollection
      category="Batman"
      key="Batman"
    />
    <GifCollection
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en <GifExpertApp/> debe mostrarse correctamente  1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
